### ISSUE TITLE
Unable to change vue-select value after setting it via model/function

### ISSUE THREAD

**Issue by JMele11968** [sagalbot/vue-select#1472]
**Describe the bug**

After setting the vue-select via model I am unable to clear the input or choose another option. I realize I may be doing this wrong, but at this point I feel like I have tried everything reasonable...

**To Reproduce**

https://codepen.io/jmele11968/pen/mdmQvQL?editors=1111

Choosing something in the top box, will set the bottom 2 and thus produce my issue being, I can not in any way un set or change this(except for if I choose a different option from the first dropdown).

**Expected behavior**

From what I can tell, the events do fire correctly, the right chosen option shows in the event log without an issue but for the life of me I can't get this to clear even programmatically. 

I have tried with/without the reduce. I have tried with a simple array as well as an array like provided where there is a label/code object. I've tried binding an event to @input and changing it via this.form.<key> = val and while it does change the form value the vue-select does not render the change or give an indication the value has actually changed.

I expect my second/third in this example select to be populated with an option, but still be changeable so another option can be selected.

**Screenshots**

codepen provided produces error without an issue

Basically looking to find out either a. what I'm doing wrong or b. what is going on and how I can solve the problem. 

I feel like setting a vue-select value in my case (or for example on a page load and database fetch) should be possible, and it should be mutable afterwards, but that doesn't seem to be the case.

**Comment by hanachan1026** [comment]
Same issue,

and maybe this is relevant

I have this kind of data

```
const options = [
    { text: "AAA", value: 1 },
    { text: "BBB", value: 2 },
]
<v-select
    id="modal-job_type"
    :value="1" // v-model also doesnt work for me
    item-value="value"
    name="job_type"
    :options="options"
    label="text"
/>
```

I expect to see "AAA" selected.

But instead I see "1" text.

The document is unclear about what value I should pass.
